staging/src/k8s.io/apiserver/pkg/util/x509metrics/missing_san.go

```diff
@@ -90,3 +157,50 @@ func hasSAN(c *x509.Certificate) bool {
 	}
 	return false
 }
+
+type sha1SignatureChecker struct {
+	*counterRaiser
+}
+
+func NewSHA1SignatureDeprecatedChecker(counter *metrics.Counter) *sha1SignatureChecker {
+	return &sha1SignatureChecker{
+		counterRaiser: &counterRaiser{
+			counter: counter,
+		},
+	}
+}
+
+// CheckRoundTripError returns true when we're running w/o GODEBUG=x509sha1=1
+// and the client reports an InsecureAlgorithmError about a SHA1 signature
+func (c *sha1SignatureChecker) CheckRoundTripError(err error) bool {
+	var unknownAuthorityError x509.UnknownAuthorityError
+	if err == nil {
+		return false
+	}
+	if !errors.As(err, &unknownAuthorityError) {
+		return false
+	}
+	if strings.Contains(err.Error(), "temporarily override with GODEBUG=x509sha1=1") {
```

Comment:
I realized this will no longer match once the fallback is dropped from go. if we want to continue detecting this, we'll need to match the `cannot verify signature: insecure algorithm %s` portion of the error. as long as unit tests catch when we bump to a go version where this is required, I don't feel strongly about changing this now or later

Reply:
Perhaps `if strIdx := strings.Index(errMsg, "x509: cannot verify signature: insecure algorithm"); strIdx != -1 && strings.Contains(errMsg[strIdx:], "SHA1")` would make more sense so that we don't have to change this in the future?

Reply:
that sounds good to me